### PR TITLE
BUG:  Holiday(..) with both offset and observance raises

### LINKED ISSUE #10217: Holidays Offset not Returning Correct Date
Why would the below return 11/2/2015 as opposed to 11/30/2015 if 11/30 would be considered the next Monday from the 4th Saturday?
    
    Holiday("Cyber Monday", month=11, day=1, offset=pd.DateOffset(weekday=SA(4)), observance=next_monday)

### PR DESCRIPTION
**Pull request by bwillers**
Resolves #10217 

Doesn't fix the underlying issue yet, but at least raises on the unsupported behavior.

### CODE REVIEW

File: pandas/tests/test_tseries.py [hunk 1]
```diff
@@ -737,6 +737,17 @@ def test_get_period_field_raises_on_out_of_range(self):
     def test_get_period_field_array_raises_on_out_of_range(self):
         self.assertRaises(ValueError, period.get_period_field_arr, -1, np.empty(1), 0)
 
+
+class TestHolidayConflictingArguments(tm.TestCase):
+
```
Comment (jreback) [inline pandas/tests/test_tseries.py]: can you add the issue number as a comment here

File: pandas/tests/test_tseries.py [hunk 2]
```diff
@@ -737,6 +737,17 @@ def test_get_period_field_raises_on_out_of_range(self):
...
+    def test_both_offset_observance_raises(self):
+        from pandas.tseries.holiday import Holiday, SA, next_monday
```
Comment (jreback) [inline pandas/tests/test_tseries.py]: these can be (maybe already are) imported at the top of the file

**Comment by jreback** [comment]
pls add a release note (put it in the API section for 0.17.0). minor changes otherwise. pls ping when green on travis.

**Comment by bwillers** [comment]
As requested. By the way, five points to whoever wrote the 'contributing to pandas' doc section.

**Comment by jreback** [comment]
thanks to 

cc @rockg who put most of that together!

**Comment by jreback** [comment]
cc @rockg

look good to you?

**Comment by rockg** [comment]
Looks good to me.  Let's keep the original issue open until that requested behavior is supported.

@bwillers Glad it was useful...hopefully removes the major hurdles to helping out.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/d513986e75110b14e288753701e435660f3d115e

thanks!

**Pull request closed without merging**